src/google/protobuf/compiler/js/js_generator.cc

```diff
@@ -2559,7 +2559,7 @@ void Generator::GenerateClassFieldFromObject(
     } else {
       printer->Print(
            "  obj.$name$ != null && jspb.Message.setField(msg, $index$, "
-           "(obj.$name$ && obj.$name$.map(v => $enumprefix$$enumname$[obj.$name$]).filter(v => v !== undefined) || undefined));\n",
+           "(obj.$name$ && obj.$name$.map(v => $enumprefix$$enumname$[v]).filter(v => v !== undefined) || undefined));\n",
```

Comment:
Basically instead of 
```js
obj.quality_issues.map(
        (v) => proto.indigofair.data.ProductQualityIssueType[obj.quality_issues]
      )
```
we need 
```js
obj.quality_issues.map(
        (v) => proto.indigofair.data.ProductQualityIssueType[v]
      )
```

Reply:
Tested locally:
![image](https://user-images.githubusercontent.com/7275468/129782890-97a2f404-c9da-4689-9ac7-0dc1b9a19853.png)

Reply:
Do you have a screenshot of the email too just to confirm?

Reply:
yep its here:
![image](https://user-images.githubusercontent.com/7275468/129784713-556e05ee-5647-48bb-95f2-ddb1e857e09d.png)

Reply:
Amazing!!! 🙏 🙏 🙏